src/backends/pgfplotsx.jl

```diff
@@ -479,22 +490,17 @@ function _update_plot_object(plt::Plot{PGFPlotsXBackend})
         for series in series_list(sp)
             for col in (:markercolor, :fillcolor, :linecolor)
                 if typeof(series.plotattributes[col]) == ColorGradient
-                    # TODO: fix this
-                    # pushed_colormap || push!(PGFPlotsX.CUSTOM_PREAMBLE, """\\pgfplotsset{
-                        # colormap={plots}{$(pgfx_colormap(series.plotattributes[col]))},
-                    # }""")
-                    pushed_colormap = true
-                    push!(axis_opt,
-                        # "colormap" => nothing,
-                        # "colormap name" => "plots",
-                    )
-
-                    # TODO: is this needed?
-                    # if sp[:colorbar] == :none
-                    #     kw[:colorbar] = "false"
-                    # else
-                    #     kw[:colorbar] = "true"
-                    # end
+                    if !pushed_colormap
+                        push!(PGFPlotsX.CUSTOM_PREAMBLE, """\\pgfplotsset{
```

Comment:
Instead of messing with the global variable, it might be better to create a `TikzDocument` and use `push_preamble!`. https://github.com/KristofferC/PGFPlotsX.jl/blob/bed8c09a1636ec884d2ad6897124ae853a8d12c7/src/tikzdocument.jl#L32

The global variables are more for end users.

Reply:
good to know, thanks! I think I need a `TikzDocument` either way to to toggle standalone `.tex` files